### ISSUE TITLE
`vim.o.mouse = nil` no longer works as of nvim 0.10.0

### ISSUE THREAD

**Issue by Samasaur1** [neovim/neovim#28992]
### Problem

In my Neovim config (`init.lua`), I have the following line:
```lua
vim.o.mouse = nil
```
When I was using nvim 0.9.5, this would properly change the value of `mouse` (verifiable with `:set mouse?` printing `mouse=`).

However, in nvim 0.10, this no longer works. `mouse` is set to `nvi` despite this line remaining in my config (verifiable with `:set mouse?` printing `mouse=nvi`). I can run `:set mouse=` in every new Neovim instance, but this seems like a regression. For what it's worth, running `:lua vim.o.mouse = nil` doesn't change anything even in a Neovim instance, although `:lua print(vim.o.mouse)` does reflect the latest change made by `:set mouse`.

### Steps to reproduce

1. Create `minimal.lua` as:
    ```lua
    vim.o.mouse = nil
    ```
2. Run `nvim --clean -u minimal.lua`
3. In Neovim, run `:set mouse?`
4. Observe that `mouse=nvi`

The same steps with Neovim 0.9.5 have `mouse=`

### Expected behavior

Neovim 0.10.0 should continue to allow you to set `mouse` using `vim.o.mouse`, or, if not, there should be a clear migration path to continue setting this configuration option in Lua

### Neovim version (nvim -v)

nvim v0.10.0

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

macOS 14.5

### Terminal name/version

alacritty 0.13.2

### $TERM environment variable

xterm-256color

### Installation

Nixpkgs

**Comment by zeertzjq** [comment]
`vim.o.mouse = ''`

**Comment by wookayin** [comment]
It seems that `nvim_set_option()` does not check or handle `nil` correctly for options with string types. While the correct usage would be `vim.o.mouse = ''`, but I think at least it should throw an error or the behaviors around `nil` should be documented.

Bisected: the change in behavior is since `e19cc9c9b7` (#25674). /cc @famiu

**Comment by famiu** [comment]
> It seems that `nvim_set_option()` does not check or handle `nil` correctly for options with string types. While the correct usage would be `vim.o.mouse = ''`, but I think at least it should throw an error or the behaviors around `nil` should be documented.
> 
> Bisected: the change in behavior is since e19cc9c9b7 (#25674). /cc @famiu 

Is `vim.o.mouse = nil` supposed to work? Is it documented behavior?

**Comment by wookayin** [comment]
No. Option value being `nil` was actually never documented (also it has never been raising any errors).

**Comment by famiu** [comment]
> No. Option value being `nil` was actually never documented (also it has never been raising any errors).

Then this is not a breaking change. The way to unset string option values was always setting them to an empty string.

**Comment by helq** [comment]
Thank you, I have fixed my config. Instead of `nil`, I used `""` :)

It would be best to show a warning to the user though. As a user who reluctantly updated their settings from .vim to .lua, I find it very frustrating to open neovim and find out that my environment does not longer work as intended after an update. Learning how to use a library (and all its nooks and cranies) on top of learning lua is tolling task. (And I thought learning how to move around the editor was the hard part XD.)

I think that helping the user with a warning, so that they don't have to spend 3 hours debugging a configuration file that worked, would be of great utility.

I know it sounds a bit like I am ranting. I am sorry, I didn't intend it. In fact, I thank you deeply for all your work on this very nice editor. Please, consider this as an enhancement. You can save hundreds of work hours :)

**Comment by famiu** [comment]
> Thank you, I have fixed my config. Instead of `nil`, I used `""` :)
> 
> It would be best to show a warning to the user though. As a user who reluctantly updated their settings from .vim to .lua, I find it very frustrating to open neovim and find out that my environment does not longer work as intended after an update. Learning how to use a library (and all its nooks and cranies) on top of learning lua is tolling task. (And I thought learning how to move around the editor was the hard part XD.)
> 
> I think that helping the user with a warning, so that they don't have to spend 3 hours debugging a configuration file that worked, would be of great utility.
> 
> I know it sounds a bit like I am ranting. I am sorry, I didn't intend it. In fact, I thank you deeply for all your work on this very nice editor. Please, consider this as an enhancement. You can save hundreds of work hours :)

While, yes, a warning would be nice. The implementation for that may require more code complexity than is necessary, especially since it's planned to change the behavior of `nil` after certain refractors are done anyway.

**Comment by tahoe** [comment]
neither works in nvim v0.11.2
but weirdly, :set mouse= from inside nvim works fine
